init.lua

```diff
@@ -1220,6 +1220,97 @@ function Form:update_where(func)
     end
 end
 
+local embed_create_ctx_mt = {}
```

Comment:
This is what I was talking about with the metatables. It has a bit of complexity, but it should be very fast.

Reply:
I haven't been able to find much information on this metatable optimisation, but I assume it makes it somewhat faster than using closures and a separate metatable for each `ctx.form` value somehow?

Reply:
Took some digging, but my sources trace back to this mailing list post from Mike Paul (creator of LuaJIT). https://lua-users.org/lists/lua-l/2011-02/msg01102.html

Reply:
Thanks, I hope it didn't take too long to find (sorry)

Reply:
nah, just a few minutes. It was well worth the hassle. Mike's emails are very entertaining and educative to read.